Source/Moya.swift

```diff
@@ -24,6 +24,45 @@ public protocol TargetType {
     var sampleData: NSData { get }
 }
 
+public enum StructTarget<T: TargetType>: TargetType {
```

Comment:
Nice one! I really like that 🙌 One thing I can add to this is that we can create a stored property for example that will give us the target, e.g.:
```swift
private var target: T {
    switch self {
        case .Struct(let t): return t
    }
}
```

and then we can skip the switch stuff in other properties with just e.g. `return target.path`.